src/server/pfs/server/driver_new_storage.go

```diff
@@ -298,24 +299,105 @@ func (fr *FileReader) drain() error {
 	return nil
 }
 
-func (d *driver) compact(master *work.Master, outputPath string, inputPrefixes []string) (retErr error) {
+// compact is the entrypoint to compaction.
+func (d *driver) compact(ctx context.Context, master *work.Master, outputPath string, inputPrefixes []string) error {
+	// resolve prefixes into paths
+	inputPaths := []string{}
+	for _, inputPrefix := range inputPrefixes {
+		err := d.storage.WalkFileSets(ctx, inputPrefix, func(inputPath string) error {
```

Comment:
This should be pulling from the master context (`m.Ctx()`), not the background. The `compact` function should not take a separate context as a parameter.

Reply:
Also, `WalkFileSets` -> `WalkFileSet`

Reply:
Also, the if statement can just operate on the results of the function call, you don't need it to be below.

Reply:
I always find it hard to read the compound if statements when they are broken up by a function literal.  What do you think?

Reply:
I kind of prefer this pattern for the simple case where I am just returning the error if it is not nil, but for more complex logic it could make sense to split it up. The general pattern in the codebase for the simple case is to just have the function call in the if statement.